legion/engine/physics/data/capsule_controller_data.hpp

```diff
@@ -0,0 +1,134 @@
+        {
+            m_modificationFlags.set(capsule_character_flag::cc_move_to, true);
+            m_currentDisplacement += displacement;
+        }
+
+        L_ALWAYS_INLINE void setRadius(float radius)
+        {
+            m_modificationFlags.set(capsule_character_flag::cc_set_radius, true);
+            m_radius = radius;
+        }
+
+        L_ALWAYS_INLINE float getRadius() const noexcept { return m_radius; }
+
+        L_ALWAYS_INLINE void setHeight(float height)
+        {
+            m_modificationFlags.set(capsule_character_flag::cc_set_height, true);
+            m_height = height;
+        }
+
+        L_ALWAYS_INLINE float getHeight() const noexcept { return m_height; }
+
+        L_ALWAYS_INLINE const math::vec3& getCurrentDisplacement() const noexcept { return m_currentDisplacement; }
+
+        L_ALWAYS_INLINE void resetDisplacement() { m_currentDisplacement = math::vec3(0); }
+
+        L_ALWAYS_INLINE const std::bitset<capsule_character_flag::cc_max>& getModificationFlags() const noexcept { return m_modificationFlags; }
+
+        L_ALWAYS_INLINE void resetModificationFlags() { m_modificationFlags.reset(); }
+
+        template<class preset>
+        pointer<const preset> getReadOnlyPreset() const
+        {
+            static size_type searchHash = typeHash<preset>();
+
+            for (const controller_preset& contPreset : m_presets)
+            {
+                if (searchHash == contPreset.hash)
+                {
+                    const void* specificsPtr = &contPreset.specifics;
+                    return { static_cast<const preset*>(specificsPtr) };
+                }
+            }
+
+            return { nullptr };
+        }
+
+        template<class preset>
+        pointer<preset> getPreset() 
+        {
+            static size_type searchHash = typeHash<preset>();
+
+            for (controller_preset& contPreset : m_presets)
+            {
+                if (searchHash == contPreset.hash)
+                {
+                    void* specificsPtr = &contPreset.specifics;
+                    return { static_cast<preset*>(specificsPtr) };
+                }
+            }
+
+            return { nullptr };
+        }
+
+        template<class preset>
+        pointer<preset> addPreset(preset& presetSpecifics)
+        {
+            static_assert(std::is_standard_layout<preset>());
+            static size_type searchHash = typeHash<preset>();
+
+            for (controller_preset& contPreset : m_presets)
+            {
+                if (searchHash == contPreset.hash)
+                {
+                    void* specificsPtr = &contPreset.specifics;
+                    return { static_cast<preset*>(specificsPtr) };
+                }
+            }
+
+            controller_preset newPreset;
+            newPreset.hash = searchHash;
+            //memcpy here would be fine since preset is of standard layout
+            memcpy(&newPreset.specifics, &presetSpecifics, sizeof(preset)); 
```

Comment:
fair point, I'll use this constexpr if way instead since there won't be a large number of presets anyway. But I think I will still keep the standard layout restriction to ensure that presets are simple